docs/DeveloperGuide.md

```diff
@@ -1,11 +1,267 @@
+  + Pros: Easy to implement.
+  + Cons: All ingredients containing the keyword will be listed. This means that even if the keyword doesn't make sense, such as supplying "to", both potato and tomato will be listed since it contains "to".
+
+* **Alternative 2**: The user supplies the exact ingredient name.
+  + Pros: The ingredient that matches exactly with the keyword supplied will be listed.
+  + Cons: The user have to input the ingredient name correctly each time he/she uses the search stock feature.
+
+##### Aspect: Data structure to support the search stock feature.
+* **Alternative 1 (current choice)**: Use a `List` to store all possible ingredient names that contain the keyword supplied.
+  + Pros: Easy to understand.
+  + Cons: The `List` is derived by converting the existing stock (which is a `HashMap`) into an ArrayList. In order to print out the ingredient's name, quantity and price, methods such as `getKey()` and `getValue()` have to be used.
+
+* **Alternative 2**: Use the existing `HashMap` data structure employed by the `stock` object.
+  + Pros: We do not need to create a new `List` to display the relevant results.
+  + Cons: Iterating a HashMap can be done by converting it to an EntrySet or by using an iterator. These, compared to `List`, is much less straightforward.
+
+<a name="list-stock"></a>
+
+### 1.2 [Proposed] List stock ingredients in descending quantity
+#### 1.2.1 Proposed implementation
+
+In the restaurant daily report, users can view all the ingredients presently in the stock by supplying the input `list stock`. The ingredients will be ordered in descending quantities, that is, the ingredient that has the highest quantity will be listed first and vice versa.
+
+Given below is an example usage scenario and how the listing mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. An empty `stock` will be initialized.
+
+Step 2. The user executes `add stock; i/tomato; q/10; p/$0.40;` command to add a tomato ingredient into the `stock`. Further, the user may add more ingredients into the current `stock`. Suppose the user executes `add stock; i/potato; q/0; p/$0.40;` and `add stock; i/rice; q/55; p/$0.40;` as well.
+
+Step 3. The user can now view the current `stock` to see what ingredients are there in the stock. The user now executes `list stock`, which will display the following result in the image. In this case, `rice` has the highest quantity of `55`, which is listed first as compared to `potato`, which has the lowest quantity of `0`. This can be seen from the image below.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/59989652/77300028-5cb97980-6d28-11ea-9a27-7d118de3431f.PNG">
+</p>
+
+The following class diagram shows how the listing operation works:
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/59989652/77920226-cc070e80-72d0-11ea-82a5-b285173d4844.png">
+</p>
```

Comment:
Perhaps a better diagram to use in this context would be a sequence diagram, since the explanation that follows this diagram seems to be about how different methods are being called and objects are being created. 

The class diagram does not really show interactions between different objects well, in that we cannot really see the order or the sequence of the interactions. All we can tell from the diagram is which object maintains associations with which other objects for example.